app/models/king.rb

```diff
@@ -1,2 +1,11 @@
 class King < Piece
+	def valid_move?(new_rank, new_file)
+		rank_diff = (new_rank - rank).abs
+		file_diff = (new_file - file).abs
+		if (rank_diff <= 1) && (file_diff <= 1) && !is_move_obstructed?(new_rank, new_file) && !is_move_malformed?(new_rank, new_file)
```

Comment:
What I mean is that `is_move_obstructed?` may do something unexpected, like throwing an exception or returning an incorrect result, if it's given a malformed move. I didn't think through all the possible scenarios when I wrote it because I expected `is_move_malformed?` to always be called first.

The other consideration is that it potentially checks a large number of squares, which makes it the most computationally expensive step in the process. That means that it's best to call it last (i.e. only if needed).

You can do this as follows using the new pattern you just suggested:
```
def valid_move?(new_rank, new_file)
  return false if is_move_malformed?(new_rank, new_file)
  return false if (new_rank - rank).abs > 1 || (new_file - file).abs > 1
  !is_move_obstructed?(new_rank, new_file)
end
```

Or you can use the original pattern:
```
if !is_move_malformed?(new_rank, new_file) && (rank_diff <= 1) && (file_diff <= 1) && !is_move_obstructed?(new_rank, new_file) 
  true			
else
  false
end
```

The key is that when you chain boolean expressions using `&&` or `||`, Ruby will start evaluating from left to right and will go as far as needed to determine the result. That means that if you have three expressions chained with `&&` and the first one evaluates to `false`, Ruby will immediately return `false` for the full chain without evaluating expressions two or three.